transaction-pool/src/pool.rs

```diff
@@ -625,12 +625,12 @@ impl<'a, T, R, S, L> Iterator for PendingIterator<'a, T, R, S, L> where
 				},
 				_ => (),
 			}
-			match tx_state {
-				Readiness::Ready => {
-					return Some(best.transaction.transaction)
-				},
-				_ => trace!("[{:?}] Ignoring {:?} transaction.", best.transaction.hash(), tx_state),
+
+			if tx_state == Readiness::Ready {
```

Comment:
I've actually meant to put the if inside previous match and leave the trace in `_` pattern, but this is fine as well.

Just got an idea that for readability we could also change both this `if` and `match` above to `if let`, but it's up to you.

Reply:
I thought it would be clearer to have one match when we need to find the next transactions, and another one when we need to return the transaction, but I'm open to anything :)

Reply:
My reasoning was that it's easier to see everything that happens in particular case(ses) in a single block, cause now you fall-through and need to analyze multiple blocks to think about what happens. But as I said, I'm not strong on this either.